src/Analyzers/CSharp/Analyzers/RemoveUnnecessarySuppression/CSharpRemoveUnnecessarySuppressionDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.RemoveUnnecessarySuppression
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class CSharpRemoveUnnecessarySuppressionDiagnosticAnalyzer : AbstractCodeStyleDiagnosticAnalyzer
+    {
+        public CSharpRemoveUnnecessarySuppressionDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.RemoveUnnecessarySuppressionForIsExpressionDiagnosticId,
+                   new LocalizableResourceString(nameof(CSharpAnalyzersResources.Remove_unnecessary_suppression_operator), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)),
+                   new LocalizableResourceString(nameof(CSharpAnalyzersResources.Suppression_operator_has_no_effect_and_can_be_misinterpreted), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)))
+        {
+        }
+
+        protected override void InitializeWorker(AnalysisContext context)
+            => context.RegisterSyntaxNodeAction(AnalyzeSyntax, SyntaxKind.IsExpression, SyntaxKind.IsPatternExpression);
+
+        private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
+        {
+            var node = context.Node;
+            var left = node switch
+            {
+                BinaryExpressionSyntax binary => binary.Left,
+                IsPatternExpressionSyntax isPattern => isPattern.Expression,
+                _ => throw ExceptionUtilities.UnexpectedValue(node),
+            };
+
+            if (left.Kind() != SyntaxKind.SuppressNullableWarningExpression)
+                return;
+
+            context.ReportDiagnostic(DiagnosticHelper.Create(
+                Descriptor,
+                ((PostfixUnaryExpressionSyntax)left).OperatorToken.GetLocation(),
+                ReportDiagnostic.Warn,
+                ImmutableArray.Create(node.GetLocation()),
+                properties: null));
```

Comment:
This seems a bit weird - the descriptor has default severity `Hidden` but analyzer always reports `Warning` diagnostic by default. Can we instead update the `AbstractCodeStyleDiagnosticAnalyzer` to take an optional default severity that defaults to hidden?

Reply:
Ok looked.  Seems unnecessary.  The descriptor is just defaulted to a simple value for everyone to use.  And all consumers just override as necessary.  Seems fine to me :)